homeassistant/helpers/script.py

```diff
@@ -515,6 +515,9 @@ def _handle_exception(
         if not isinstance(exception, exceptions.HomeAssistantError):
             raise exception
 
+        # Store the error as variable for future steps.
+        self._variables["last_error"] = f"Error: {exception}"
```

Comment:
I think we should not use a variable for that, but use actual logic instead. 

For example, consider the introduction of a try/except or the ability to set a sequence of actions to run when an error happens (which in both cases passes the error).

Reply:
Many thanks for the feedback!

I'll explore enhancing the script with a try/except step.